### DISCUSSION TITLE
No longer relies on API module programming in some scenarios

### DISCUSSION THREAD

**Discussion by Aresxue** [apache/dubbo#13890] [General]
## Background
When we need to call a dubbo service, we usually need to introduce dependency xxxx-api, but in some cases (such as gateways) the scalability is very poor. Today's metadata center is gradually becoming mature, so we hope to explore In some scenarios, metadata can be reasonably used to replace the programming prompt function of xxxx-api.

## Describe the proposal
The metadata center already has relevant information about the interface. Here is a simple implementation idea to implement programming prompts through IDE plug-ins.
First we select the application that needs to consume the service, which will show us the interfaces it has. Then we select the interface we need (confirmed version and group) through retrieval or smart prompts, and then select its method. After selecting the method It will automatically prompt the input and output parameters to complete the programming process. After this, you can directly generate the generalized calling code. It is best to perform this replacement in the compiler. What you see in the IDE is the code programmed by yourself, which looks like the previous calling method.

## Goals
- No longer relies on API module programming in some scenarios

## Non-Goals
- Completely replace the original programming method

**Comment by AlbumenJ** [comment]
Actually I don't know what you want for Dubbo itself. A new programming style for Dubbo developers or Dubbo users?

**Comment by Aresxue** [comment]
> Actually I don't know what you want for Dubbo itself. A new programming style for Dubbo developers or Dubbo users?

Sorry, I kind of lost the point.For dubbo itself, I expect to provide the function of converting ordinary java calling code into generalized calling code. This may already exist but I did not find it. In addition, are you interested in making a small code editor or an idea plugin?
